src/realm/bplustree.cpp

```diff
@@ -769,3 +833,14 @@ std::unique_ptr<BPlusTreeNode> BPlusTreeBase::create_root_from_ref(ref_type ref)
         return new_root;
     }
 }
+
+size_t BPlusTreeBase::size_from_header(const char* header)
+{
+    auto node_size = Array::get_size_from_header(header);
+    if (Array::get_is_inner_bptree_node_from_header(header)) {
+        auto data = Array::get_data_from_header(header);
+        auto width = Array::get_width_from_header(header);
+        node_size = size_t(get_direct(data, width, node_size - 1)) >> 1;
+    }
+    return node_size;
```

Comment:
```suggestion
    if (Array::get_is_inner_bptree_node_from_header(header)) {
        auto data = Array::get_data_from_header(header);
        auto width = Array::get_width_from_header(header);
        auto node_size = size_t(get_direct(data, width, node_size - 1));
        REALM_ASSERT_EX(node_size >= 2, node_size);
        return node_size >> 1;
    }
    return Array::get_size_from_header(header);
```

Reply:
Also, double checking because my memory of this is fuzzy, is the divide by two always correct for an inner node? Or is it possible that it is not in the compact form. Eg should we divide by `get_elems_per_child()`

Reply:
It is more a shift right than a divide by two. The size of the sub tree is always stored as the last entry and coded as a number - that is shifted one up and or-ed with 1.

Reply:
Not blocking, but you may want to use this suggestion to avoid an unnecessary read from the array header if looking at an inner node.

Reply:
I just don't think your suggesting would compile :-)